src/backoff.rs

```diff
@@ -6,7 +6,7 @@ use tracing::info;
 /// Exponential backoff with jitter
 ///
 /// See <https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/>
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Copy)]
```

Comment:
Deriving copy like this is a very deliberate choice, are you sure this is semantically correct?

Reply:
I'm not sure I understand this.

Reply:
Why did you derive `Copy`? Does it make sense?

A type that is copy is usually a type that is

 - Constant
 - Somehow primitive

Deriving copy means that you don't ever want the user to care whether they are cloning it. Does that make sense here? If a type should not _obviously_ be copy, then it is generally better to just force users to clone it.

Reply:
I get it now. I just stuck with the suggestions by the compiler. I think it'll be better to, probably, use an `Arc` of `BackoffConfig` instance here.

Reply:
That is _not_ what I'm suggesting. I think you can safely clone here, using an `Arc` is only relevant if you really, really don't want to copy.

It's a question of degree:

 - Very expensive or impossible to copy, or uses interior mutability -> Arc/Rc
 - Normal types -> Clone
 - Special primitive types -> Copy

Reply:
I've tried this with just clone and clippy still suggests that `Copy` be implemented on `BackoffConfig`.

Reply:
Based on [this](https://github.com/cognitedata/rskafka/blob/fe0cc3b0a447328cfa51fcb93a08610977c7c742/src/lib.rs#L11) though, it looks like they made it so intentionally.

Reply:
Alright, though I'm not sure I agree with the lint. It seems to exist due to some legacy stuff.